### PR TITLE
Scatter chart component

### PR DESCRIPTION
**Pull request by pyyding**
## Scatter chart component 

WIP

- [x]  revert tooltip changes and create a composable tooltip
- [x] agree on what to do with disabled tooltip animations 
- [x] agree on where to put shared constants 

other thoughts

- maybe should re-arrange chart stuff to a separate components/chart dir and organize component exports inside it's index files-s

```
// folder tree:
- components/
-- main.ts
-- ReferenceLine/
----/ ReferenceLine.tsx
--- / index.ts
--- / constants.ts

// ReferenceLine/index.ts would look like:
  export * from './constants'
  export { ReferenceLine } from './ReferenceLine'

// main.ts:
export * from './chart/ReferenceLine'
export * from './chart/ChartTooltip'
```

### CODE REVIEW

File: lib/components/XAxis/constants.ts [hunk 1]
```diff
@@ -1,3 +1,10 @@
+import { type XAxisProps } from "recharts";
+
+/**
+ * todo: verify - maybe move all chart components to a separate folder, add it to ui/components/chart/constants.ts
```
Comment (pyyding) [inline lib/components/XAxis/constants.ts]: @ferrucc-io does it sound good to you? keep it like that for now, and make a separate PR to re-arrange stuff?

File: src/stories/AreaChart.stories.tsx [hunk 2]
```diff
@@ -53,14 +54,24 @@ export const Default: Story = {
         <YAxis {...defaultYAxisProps} dataKey="y" />
         <XAxis {...defaultXAxisProps} dataKey="x" />
         <ReferenceLine {...defaultReferenceLineProps} x="Wed" />
-        <ChartTooltip 
-            content={({ active, payload, label }) => <DefaultTooltip label={label} active={active} payload={payload} valueFormatter={(p)=> `${p.y} count`} />}
+        <ChartTooltip
+          {...defaultTooltipProps}
```
Comment (pyyding) [inline src/stories/AreaChart.stories.tsx:59]: {...defaultTooltipProps} is a breaking change.

Three possible ways forward:
a) keep the breaking change, be sure to update it in the client
b) make an exception for `ChartTooltip` to use extended class with defaultProps.
c) try to refactor everything to class extensions with default props

i'd go with a)

File: lib/components/ReferenceLine/constants.ts [hunk 3]
```diff
@@ -4,3 +4,13 @@ export const defaultReferenceLineProps: ReferenceLineProps = {
   stroke: "inherit",
   className: "stroke-gray-300",
 };
+
+export const scatterReferenceLineXProps: ReferenceLineProps = {
```
Comment (pyyding) [inline lib/components/ReferenceLine/constants.ts:8]: We could've created separate `ReferenceLine`, etc. components for `ScatterChart` but I think this makes more sense. 

when recharts v3 is around we're gonna do something like this  
```
-- <ReferenceLine {...scatterReferenceLineXProps} />
++ <ReferenceLine variant="scatter" axis="x" />
```

Comment (ferrucc-io) [reply lib/components/ReferenceLine/constants.ts:8]: I don't think we use the reference line for Scatterplots btw

File: lib/components/XAxis/constants.ts (hunk 1 above)
Comment (ferrucc-io) [reply lib/components/XAxis/constants.ts]: yeah sounds good to me!

File: src/stories/AreaChart.stories.tsx (hunk 2 above)
Comment (ferrucc-io) [reply src/stories/AreaChart.stories.tsx:59]: We can keep the breaking change it should be cheap to fix!

File: lib/components/ReferenceLine/constants.ts (hunk 3 above)
Comment (ferrucc-io) [reply lib/components/ReferenceLine/constants.ts:8]: Taking it back, I think this makes sense!

**Review by ferrucc-io — APPROVED** [review]
nice one this one almost looks ready to go 🔥

**Review by ferrucc-io — APPROVED** [review]
